tls/s2n_psk.h

```diff
@@ -55,6 +41,8 @@ struct s2n_psk {
     uint32_t obfuscated_ticket_age;
     struct s2n_blob early_secret;
 };
+S2N_RESULT s2n_psk_init(struct s2n_psk *psk, s2n_psk_type type);
```

Comment:
Why is only `s2n_psk_init` returning S2N_RESULT? Aren't we using this within `s2n_connection_set_external_psks` API as well?

Reply:
s2n_psk_init isn't intended to be a public API. It's not a useful public API because s2n_psk is opaque, and therefore can't be used via the public API without calling s2n_external_psk_new to allocate the right amount of memory.

It's not the methods called inside s2n_connection_set_external_psks that have to use posix return types, it's the APIs intended to be public.